security-profile/oci-profile.go

```diff
@@ -0,0 +1,195 @@
+package security_profile
+
+import (
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"reflect"
+)
+
+var OciProfileType = "oci-profile"
+
+type AppArmorProfile struct {
+	Rules []string
+}
+
+// Profile maintains some OCI spec settings but should also contain a complete security
+// context.
+// Profiles should be maintained for both Linux and Windows at any given time.
+// FIXME: Add error handling here if profile or subfields are not allocated */
+// Fixme add api access settings for Engine / Swarm / K8s?
+type OciProfile struct {
+	Oci      *specs.Spec
+	AppArmor *AppArmorProfile
+}
+
+func NewOciProfile(ociSpec *specs.Spec) *OciProfile {
+	return &OciProfile{Oci: ociSpec}
+}
+
+func (p *OciProfile) GetType() string {
+	return OciProfileType
+}
+
+/* Add a list of capabilities if not present to all capability masks */
+func (p *OciProfile) AddCaps(capsToAdd ...string) {
+	for _, cap := range capsToAdd {
+		p.Oci.Process.Capabilities.Bounding = addCapToList(p.Oci.Process.Capabilities.Bounding, cap)
+		p.Oci.Process.Capabilities.Effective = addCapToList(p.Oci.Process.Capabilities.Effective, cap)
+		p.Oci.Process.Capabilities.Inheritable = addCapToList(p.Oci.Process.Capabilities.Inheritable, cap)
+		p.Oci.Process.Capabilities.Permitted = addCapToList(p.Oci.Process.Capabilities.Permitted, cap)
+
+		// Should be updated automatically if the previous masks are set
+		p.Oci.Process.Capabilities.Ambient = addCapToList(p.Oci.Process.Capabilities.Ambient, cap)
```

Comment:
As far as I understood the code I dug in, `Ambient` is supposed to be updated automatically and we do add caps to every set besides Ambient in our setup of the default OCI config: https://github.com/moby/moby/blob/930e6896685f614d8f15ed10207596b1a151cbc8/oci/defaults.go#L121